src/util/taproot.rs

```diff
@@ -718,6 +716,23 @@ impl TaprootMerkleBranch {
     }
 }
 
+impl TryFrom<Vec<sha256::Hash>> for TaprootMerkleBranch {
+    type Error = TaprootError;
+
+    /// Creates a merkle proof from list of hashes.
+    ///
+    /// # Errors
+    ///
+    /// If inner proof length is more than [`TAPROOT_CONTROL_MAX_NODE_COUNT`] (128).
+    fn try_from(v: Vec<sha256::Hash>) -> Result<Self, Self::Error> {
+        if v.len() > TAPROOT_CONTROL_MAX_NODE_COUNT {
+            Err(TaprootError::InvalidMerkleTreeDepth(v.len()))
+        } else {
+            Ok(TaprootMerkleBranch(v))
+        }
+    }
+}
```

Comment:
It'd be useful to add `TryFrom<&[sha256::Hash]>` since this is where `TryFrom` shines - efficiently handling both owned and non-owned conversions.

Reply:
I didn't have this in mind when using `TryFrom`, thanks for the lesson. Added as suggested.